Sources/SwiftDiagnostics/GroupedDiagnostics.swift

```diff
@@ -133,6 +133,32 @@ extension GroupedDiagnostics {
     }
   }
 
+  // Find the "primary" diagnostic that will be shown at the top of the diagnostic
+  // message. This is typically the error, warning, or remark.
+  private func findPrimaryDiagnostic(in sourceFile: SourceFile) -> (SourceFile, Diagnostic)? {
```

Comment:
Can a source file only have a single primary diagnostic? What if it has one error on line 10 and one on line 200?

Also: If there is a warning in line 5 and an error in line 7, I think the error should be the primary diagnostic, not the warning.

Reply:
The paired change in the Swift compiler ensures that there is at most one non-note "primary diagnostic" per call to the diagnostic formatter. We can decide what to do for any clients that don't ensure this, which could be to take the most severe diagnostic or the first non-note diagnostic or something else.